eth/filters/bor_filter_test.go

```diff
@@ -0,0 +1,164 @@
+package filters
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	types "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+
+	gomock "github.com/golang/mock/gomock"
+)
+
+var (
+	key1, _ = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	addr    = crypto.PubkeyToAddress(key1.PublicKey)
+
+	hash1 = common.BytesToHash([]byte("topic1"))
+	hash2 = common.BytesToHash([]byte("topic2"))
+	hash3 = common.BytesToHash([]byte("topic3"))
+	hash4 = common.BytesToHash([]byte("topic4"))
+)
+
+func newTestHeader(blockNumber uint) *types.Header {
+	head := types.Header{
+		Number: big.NewInt(int64(blockNumber)),
+	}
+
+	return &head
+}
+
+func newTestReceipt(contractAddr common.Address, topicAddress common.Hash) *types.Receipt {
+	receipt := types.NewReceipt(nil, false, 0)
```

Comment:
if you have the same logs for all receipts, then you might not catch issues in the code.

Reply:
The logs differ, the topic address is different for every log.

Reply:
https://github.com/maticnetwork/bor/blob/58ef6fdbb7ecc37a3eeebd0c0410917b00268327/eth/filters/bor_filter_test.go#L56

Reply:
I see now.  `backend.EXPECT().GetBorBlockReceipt(gomock.Any(), gomock.Any()).Return(newTestReceipt(addr, *hashes[i]), nil)` hashes doesn't stand for hash, but rather for topics.
No questions then